### DISCUSSION TITLE
render plugin options

### DISCUSSION THREAD

**Discussion by studiopatagon** [jeremyevans/roda#309] [General]
Hi there, I was looking to enable whitespace removal from erb, but found it easier to just switch to erubis instead. The only issue is that it'll only work if all my templates have .erubis extension. Is there any way to have it take .erb? :)

Thanks in advance!

**Comment by jeremyevans** [comment]
You probably don't want to use erubis, it's not maintained.  tilt (which Roda's render plugin uses) will use erubi if it is installed, and erubi supports the same trim options as erubis.  tilt will use erubi for the .erb file extension if erubi is installed.

**Reply by studiopatagon** [reply]
Thanks! I installed erubi and it's now asking me for .erubi extension... It works great though as far as white space... is there a way to set the extension?

**Reply by janko** [reply]
That's strange, using `.erb` has always worked for me. What is your render plugin configuration?

**Reply by jeremyevans** [reply]
I'm not sure what you mean by `it's now asking me for .erubi extension`.  While the `.erubi` extension is specific to erubi, tilt will use erubi if availble for both the `.erb` and `.rhtml` extensions by default, since erubi has the highest priority of the erb libraries that tilt supports.

**Reply by studiopatagon** [reply]
I'm doing this

`plugin :render, engine: 'erubi'`

But I've also tried:

`plugin :render, :escape=>true`

The above causes my HTML tags to be escaped...

**Reply by jeremyevans** [reply]
If you are specifying `engine: 'erubi'`, then it will look for the `.erubi` file extension by default.  You shouldn't need to specify the engine.  The default engine is `erb`, and as already stated, tilt will use erubi for `.erb` file extensions if erubi is installed.

**Reply by studiopatagon** [reply]
Got it, makes sense. Since I'm new to this, is there any way to have verbose output so I know what's being loaded? 

Also, is the showing of HTML tags when using escape=>true the expected behavior?

**Reply by jeremyevans** [reply]
There's no verbose output by default.  But the basic path is: `#{views_dir}/#{template_name}.#{template_engine}`.  The default `views_dir` is 'views', and the default `template_engine` is `erb`.

When using `:escape=>true`, `<%=` ERB tags will escape , and `<%==` ERB will not escape.  So if you have an expression you expect to return valid HTML, you should use `<%==`.  For everything else, use `<%=`.  I run all of my applications with `:escape=>true`.

**Reply by studiopatagon** [reply]
Got it, thanks! It's all working accordingly now. I'll get you back someday for all this help.
